pkg/tc/actuator_tc.go

```diff
@@ -0,0 +1,108 @@
+package tc
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+
+	"github.com/Mellanox/multi-networkpolicy-tc/pkg/tc/types"
+)
+
+// NewActuatorTCImpl creates a new ActuatorTCImpl
+func NewActuatorTCImpl(tcIfc TC, log klog.Logger) *ActuatorTCImpl {
+	return &ActuatorTCImpl{tcApi: tcIfc, log: log}
+}
+
+// ActuatorTCImpl is an implementation of Actuator interface using provided TC interface to apply TC objects
+type ActuatorTCImpl struct {
+	tcApi TC
+	log   klog.Logger
+}
+
+// Actuate is an implementation of Actuator interface. it applies TCObjects on the representor
+// Note: it assumes all filters are in Chain 0
+func (a *ActuatorTCImpl) Actuate(objects *TCObjects) error {
+	// list qdiscs
+	currentQDiscs, err := a.tcApi.QDiscList()
+	if err != nil {
+		return errors.Wrap(err, "failed to list qdiscs")
+	}
+
+	var ingressQDiscExist bool
+	for _, q := range currentQDiscs {
+		if q.Type() == types.QDiscIngressType {
+			ingressQDiscExist = true
```

Comment:
Do we need to add a break statement here?

Reply:
yes, will add.